src/core/EdFi.DataManagementService.Core/Backend/PostgresqlDocumentStoreRepository.cs

```diff
@@ -3,51 +3,82 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
-using EdFi.DataManagementService.Core.Model;
 using System.Text.Json.Nodes;
 using Dapper;
+using EdFi.DataManagementService.Core.Model;
 using Microsoft.Extensions.Logging;
 using Npgsql;
 
 namespace EdFi.DataManagementService.Core.Backend;
 
-public class PostgresqlDocumentStoreRepository(ILogger<PostgresqlDocumentStoreRepository> _logger, string _connectionString)
-    : IDocumentStoreRepository, IQueryHandler
+public class PostgresqlDocumentStoreRepository(
+    ILogger<PostgresqlDocumentStoreRepository> _logger,
+    string _connectionString
+) : IDocumentStoreRepository, IQueryHandler
 {
+    // Returns an integer in the range 0..15 from the last byte of a DocumentUuid
+    private static int PartitionKeyFor(DocumentUuid documentUuid)
+    {
+        Guid asGuid = Guid.Parse(documentUuid.Value);
+        byte lastByte = asGuid.ToByteArray()[^1];
+        return lastByte % 16;
```

Comment:
Modulo the last byte of the uuid is simple to explain

Reply:
While the function is relatively trivial, I would still like to see some unit tests for it.

Reply:
Tests added. Moved the algo to a common base class for all repositories that need partitioning.